git/git.go

```diff
@@ -1080,6 +1080,45 @@ func RemoteRefs(remoteName string) ([]*Ref, error) {
 	return ret, cmd.Wait()
 }
 
+// AllRefs returns a slice of all references in a Git repository, or an error if
+// those references could not be loaded.
+func AllRefs() ([]*Ref, error) {
+	cmd := subprocess.ExecCommand("git",
+		"for-each-ref", "--format='%(objectname)%00%(refname)'")
```

Comment:
Using a NUL here is probably overkill. The objectname can't have any whitespace in it. It looks like splitting on NUL isn't any harder, so it's fine as an extra precaution.

The entries are still delimited by newlines, but that's OK, as newlines are forbidden in refnames.